File: Example/Example/Post.m
//
//  Post.m
//  JSONObjectMapperExample
//
//  Created by Sergey on 20/05/2018.
//  Copyright © 2018 Sergey. All rights reserved.
//

#import <JSONObjectMapper/JSONObjectMapping.h>
#import <JSONObjectMapper/JSONRelationshipMapping.h>
#import "Post.h"

@implementation Post

@dynamic postID;
@dynamic content;
@dynamic rating;
@dynamic viewCount;
@dynamic author;
@dynamic comments;

+ (JSONObjectMapping *)defaultMapping {
    return [JSONObjectMapping mappingWithDictionary:@{
        @"postID": @"post_id",
        @"content": @"content",
        @"rating": @"rating",
        @"viewCount": @"view_count",
        @"author": [JSONRelationshipMapping mappingWithJSONKey:@"author_id"],
        @"comments": [JSONRelationshipMapping mappingWithJSONKey:@"comments"]
    }];
}

+ (NSString *)primaryKey {
    return @"postID";
}

@end

File: ExampleSwift/ExampleSwift/Post.swift
//
//  Post.swift
//  JSONObjectMapper_Example
//
//  Created by Sergey on 21/08/2018.
//

import CoreData
import UIKit

@objc(Post)
public class Post: NSManagedObject, JSONMappingProtocol {

    @NSManaged var postID: NSNumber
    @NSManaged var content: String?
    @NSManaged var rating: NSNumber?
    @NSManaged var viewCount: NSNumber?
    @NSManaged var author: User?
    @NSManaged var comments: Set<Comment>
    
    public class func defaultMapping() -> JSONObjectMapping! {
        return JSONObjectMapping(dictionary: [
            "postID": "post_id",
            "content": "content",
            "rating": "rating",
            "viewCount": "view_count",
            "author": JSONRelationshipMapping(jsonKey: "author_id"),
            "comments": JSONRelationshipMapping(jsonKey: "comments"),
        ])
    }
    
    public class func primaryKey() -> String! {
        return "postID"
    }
}
